### PR TITLE
backend/test: refactor eth coin test using testify/suite

### PR DESCRIPTION
**Pull request by Beerosagos**

### CODE REVIEW

File: backend/coins/eth/coin_test.go [hunk 1]
```diff
@@ -35,38 +41,46 @@ func TestCoin(t *testing.T) {
 		nil,
 		nil,
 	)
+}
 
-	require.Equal(t,
+func TestSuite(t *testing.T) {
+	suite.Run(t, new(testSuite))
```
Comment (benma) [inline backend/coins/eth/coin_test.go:47]: maybe run the suite multiple times, once with ETH and once with an ERC20 token /with different number of decimals)? then the SetAmount test also covers tokens and `isFee = true` behaves differently.

**Review by benma — APPROVED** [review]
LGTM!

File: backend/coins/eth/coin_test.go (hunk 1 above)
Comment (Beerosagos) [reply backend/coins/eth/coin_test.go:47]: Ok! I'll push an update

**Comment by Beerosagos** [comment]
Added erc20 token test

**Pull request merged**